lightningd/Makefile

```diff
@@ -39,6 +39,7 @@ LIGHTNINGD_LIB_SRC :=				\
 	lightningd/channel.c			\
 	lightningd/channel_config.c		\
 	lightningd/commit_tx.c			\
+	lightningd/connection.c			\
```

Comment:
We should probably rename this daemon_conn.c/h, but again, that's pretty simple.

This overlaps my lightningd/msg_queue, but we can move this on top of that.  I'll use this to do status, too, since that's currently synchronous and mixes badly (could cause corruption) with async daemons.